### PR TITLE
[bug-fix] Fix lockfile rewrite

### PR DESCRIPTION
**Pull request by mikeland73**
## Summary

The source of this bug is that when calling `lockfile.Tidy()` we were passing in only top level packages and not including plugin packages as well.

This bug was also affecting dev container construction and some edge case plugin template logic. This bug only affects users that are using custom plugins with packages.

I renamed `PackageNames` to `AllPackageNames` for clarity. Also removed a redundant function in config package that was no longer needed.
 
## How was it tested?

In other project added golang monorepo plugin and logged to ensure Tidy was not recreating lockfile. Also tested adding go to that project.

### CODE REVIEW

File: internal/devbox/devbox.go [hunk 1]
```diff
@@ -1039,11 +1039,14 @@ func (d *Devbox) flakeDir() string {
...
-	// Skipping for now to avoid propagating the error value.
-	return d.cfg.PackagesVersionedNames()
+// AllPackageNames returns the all package names, including those added
+// by plugins
+func (d *Devbox) AllPackageNames() []string {
+	result := make([]string, 0, len(d.cfg.Packages()))
+	for _, p := range d.cfg.Packages() {
+		result = append(result, p.VersionedName())
+	}
+	return result
```
Comment (savil) [inline internal/devbox/devbox.go:1049]: Are the names from `AllPackageNames` exactly matching the set of packages in `AllPackages` functions below?

`AllPackages` seems to respect `__remove_trigger_package`. Does `AllPackageNames` do that as well?

File: internal/plugin/plugin.go [hunk 2]
```diff
@@ -144,7 +144,7 @@ func (m *Manager) createFile(
 		"DevboxDirRoot":        filepath.Join(m.ProjectDir(), devboxDirName),
 		"DevboxProfileDefault": filepath.Join(m.ProjectDir(), nix.ProfilePath),
 		"PackageAttributePath": attributePath,
-		"Packages":             m.PackageNames(),
+		"Packages":             m.AllPackageNames(),
```
Comment (savil) [inline internal/plugin/plugin.go:147]: 🤔 is this one safe to do?

**Review by savil — APPROVED** [review]

File: internal/devbox/devbox.go (hunk 1 above)
Comment (mikeland73) [reply internal/devbox/devbox.go:1049]: Good catch. Not sure. I'll audit.

File: internal/plugin/plugin.go (hunk 2 above)
Comment (mikeland73) [reply internal/plugin/plugin.go:147]: yeah this is all good.

**Review by gcurtis — APPROVED** [review]

File: internal/devbox/devbox.go (hunk 1 above)
Comment (mikeland73) [reply internal/devbox/devbox.go:1055]: Definitely needed in most places.  Added back in. I think we can improve on this (some callers want different representations of this list) but for now the safest fix is to just include everything (top level, plugin packages, and also trigger removed)

File: internal/boxcli/global.go [hunk 3]
```diff
@@ -61,33 +59,33 @@ func addCommandAndHideConfigFlag(parent, child *cobra.Command) {
 	_ = child.Flags().MarkHidden("config")
 }
 
-func globalListCmd() *cobra.Command {
-	return &cobra.Command{
+type listCmdFlags struct {
```
Comment (savil) [inline internal/boxcli/global.go:62]: why is this in this PR?

**Pull request merged**